src/connection/impls/file.rs

```diff
@@ -361,12 +363,26 @@ impl IpmiConnection for File {
                 log::debug!("Received response after {} ms", duration);
                 recv.log(log::Level::Trace);
 
-                recv.try_into().map_err(|e| {
-                    io::Error::new(
+                match Response::try_from(recv) {
+                    Ok(response) => {
+                        if response.seq() == self.seq {
```

Comment:
The only downside I see with this is that we "no longer" (not that I think anyone was ever using it, but you know) support reception of messages out-of-order.

I don't, however, think we need to worry about that now, just a comment for in the future. It should be easy to fix at any rate :)